code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -9677,26 +9677,47 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
 /datum/reagent/ectoplasm
 	name = "Ectoplasm"
 	id = ECTOPLASM
-	description = "Pure, distilled spooky"
+	description = "Pure, distilled spooky."
 	reagent_state = REAGENT_STATE_LIQUID
 	color = "#21d389b4"
 	density = 0.05
-
+	custom_metabolism = 0.01
+	var/spookvision = FALSE
+	
 /datum/reagent/ectoplasm/on_mob_life(var/mob/living/M)
 	if(..())
 		return 1
+
+	if(tick >= 30 && volume >= 1 && !spookvision) //ghostsight after 1m and having more than 1u inside
```

Comment:
put spookvision check first to save operations (extremely minor optimization, not worth holding the PR over but nice to have if you're fixing other stuff)